src/oauth-facade.ts

```diff
@@ -101,6 +101,15 @@ export default class OauthFacade {
     // Fetch a valid auth response (or throw)
     const authRequestResponse = await this.authHandler.resolveAuthorizationRequest();
 
+    // We have discovered in practice that @openid/appauth fails to properly clean up its
+    // storage entries in some cases. We add this hack to manually garbage collect all
+    // outstanding storage entries that the upstream library has left behind. Note that this
+    // depends on implementation details of both this class (namely the use of LocalStorageBackend)
+    // as well as of @openid/appauth (namely its storage key name conventions).
+    Object.keys(localStorage).filter((key) => key.includes('appauth_authorization')).forEach((key) => {
+      localStorage.removeItem(key);
+    });
```

Comment:
One alternative design would be to extend `LocalStorageBackend` and add a `cleanup` method that we could call here, and I'd be OK with doing that, but IMO the larger source of hackyness here is the reliance on the private string values from upstream.

Reply:
That sounds like we'd have exactly this snippet, but in a method on a `LocalStorageBackend` subclass, which would be the same level of hacky but also more obfuscated.

Reply:
That was my thought too.

Reply:
`LocalStorageBackend` already supports general value removal. However, the architecture of `LocalStorageBackend` is key-agnostic, so even a subclass should not perform the kind of fuzzy removal that's implemented here.

`RedirectRequestHandler` is the class that's responsible for both determining the key and for cleaning up its `storageBackend`.